packages/js/src/settings/routes/templates/taxonomy.js

```diff
@@ -132,6 +132,22 @@ const Taxonomy = ( { name, label, postTypes: postTypeNames, showUi } ) => {
 		/>;
 	}, [ name, stripCategoryBaseDescription ] );
 
+	const taxonomyNameMessage = useMemo( () => {
+		return createInterpolateElement(
+			sprintf(
+				/**
+				 * translators: %1$s expands to the name of the taxonomy.
+				 */
+				__( "The name of this category is %1$s.", "wordpress-seo" ),
+				"<link />"
+			), {
+				link: <Link href={ `edit-tags.php?taxonomy=${ name }` }>
```

Comment:
I think you might want to pass a URL from the backend. In case some funky admin URL changing is going on?
Wrapping it in `admin_url()` at the very least?